tuxemon/technique.py

```diff
@@ -297,25 +297,38 @@ def changestats(self, user: Monster, target: Monster) -> EffectResult:
         '''
         statsmaster = [self.statspeed, self.stathp, self.statarmour, self.statmelee, self.statranged, self.statdodge]
         statslugs = ['speed', 'hp', 'armour', 'melee', 'ranged', 'dodge']
+        newstatvalue = 0
         for stat, slug in zip(statsmaster, statslugs):
             if not stat:
                 continue
-            value = stat['value']
-            dividing = stat['dividing']
+            minvalue = stat['minvalue']
+            maxvalue = stat['maxvalue']
+            iterator = stat['iterator']
             override = stat['overridetofull']
             basestatvalue = getattr(target, slug)
-            if value > 0 and override == False:
-                if dividing == False:
-                    newstatvalue = basestatvalue + value
-                else:
+            if maxvalue:
+                value = random.randint(minvalue,maxvalue)
+            else:
+                value = minvalue
+            if value > 0 and  override == False:
+                if iterator == '/':
                     newstatvalue = basestatvalue // value
-                success = True
-                setattr(target, slug, newstatvalue)
+                    setattr(target, slug, newstatvalue)
+                elif iterator == '+':
+                    newstatvalue = basestatvalue + value
+                    setattr(target, slug, newstatvalue)
+                elif iterator == '-':
+                    newstatvalue = basestatvalue - value
+                    setattr(target, slug, newstatvalue)
+                elif iterator == '*':
+                    newstatvalue = basestatvalue * value
+                    setattr(target, slug, newstatvalue)
             if override == True and slug == 'hp':
-                success = True
                 target.current_hp = target.hp
+                newstatvalue = target.current_hp
+                setattr(target, slug, newstatvalue)
         return {
-            "success": bool(stat)
+            "success": bool(newstatvalue)
```

Comment:
I do not understand this success value.

Reply:
i did this because the prorgam returns that an attack failed, even if it hit,  when even one stat is 0. By putting newstatvalue=0 right before the loop, and newstatvalue in the loop once a stat is above 0, it checks for at least one stat being successful rather than all of them.

Reply:
then it fails because a stat cant be 0

Reply:
What about hp?

Reply:
yeah, current_hp is a problem.  if it is set to 0, then it is a "success".  you should check if there is a change from the old stat to the new one, then check for success.  as is, if a stat change KOs a monster, then the dialog will say it failed.

Reply:
fixed by checking if the stat is hp and if so the newstatvalue is equal to 1 which is always true

Reply:
what about the other stats?  why are you so sure they will never be 0?

Reply:
im honestly unsure why i thought that, i added a check that makes the value always at least 1 before it returns the value

Reply:
ok, just calrifing then, you are aware that this will never be false?  i see that `statdodge` has a min of 0 in the JSON.  consider if the dodge value is 3 at the start, then we get 0 for the diff.  the final result will be 3, which your function will call a success.  is that what you intend to do?  ideally, you would take the difference of the old and new, so you can reliably tell if the technique did anything or not.  because as-is, it ill always register a success, even if there was no change to the stats.

Reply:
i think itll be fine as-is, as unless the move is coded badly and somehow affects stats but has no stat changes, theres no reason as to why the stats wouldn't change